src/common/cockpitpipe.c

```diff
@@ -372,6 +374,15 @@ dispatch_input (gint fd,
   return TRUE;
 }
 
+static void
+forward_error (CockpitPipe *self)
+{
+  CockpitPipePrivate *priv = cockpit_pipe_get_instance_private (self);
+
+  fwrite (priv->err_buffer->data, priv->err_buffer->len, 1, stderr);
```

Comment:
uh... okay, whatever :rofl:

Reply:
> Even though it's not included in the length of the GBytes, the string from cockpit_unicode_force_utf8() comes from GString and is therefore already always nul terminated.

Not always.  When `g_utf8_validate` is happy with everything, the function returns the input GBytes unchanged (but with one more ref).

Reply:
I have pushed a FIXUP that works directly with nul characters instead of using the "x" as a placeholder.